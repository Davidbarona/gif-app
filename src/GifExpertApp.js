import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";


const GifExpertApp = ({defaultCategories=[]}) => {
  
  //const [ cat, setCat] = useState(['one punch']);
    const [ cat, setCat] = useState( defaultCategories );

    // const handleAdd = () =>{

    //      setCat([...cat,'Baki Hanma '])  
    //     //setCat(cats =>[...cats,'Baki'])

    // }


    return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={ setCat }/>
      <hr />

      

      <ol>
          {
              cat.map(category=>
                 <GifGrid 
                 key={category}
                 category={category}
                 />
              )
          }
      </ol>
    </>
  );
};

export default GifExpertApp;

 