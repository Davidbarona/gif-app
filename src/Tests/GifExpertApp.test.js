import React from 'react'
import '@testing-library/jest-dom'
import {shallow} from 'enzyme'
import GifExpertApp from '../GifExpertApp'

describe('test on GitExpertApp',()=>{
    
    test('should render GifExpertApp Correctly', () => { 
        const wrapper = shallow(<GifExpertApp />)
        expect(wrapper).toMatchSnapshot()
    })

    test('should show a list categories', () => {

        const categories =['one punch','dragon ball z']
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>)

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('GifGrid').length).toBe(categories.length)
     })
     

    
})