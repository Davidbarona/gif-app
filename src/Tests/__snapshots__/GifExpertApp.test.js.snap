// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on GitExpertApp should render GifExpertApp Correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`test on GitExpertApp should show a list categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="one punch"
      key="one punch"
    />
    <GifGrid
      category="dragon ball z"
      key="dragon ball z"
    />
  </ol>
</Fragment>
`;
